app/models/stage.rb

```diff
@@ -92,10 +92,15 @@ def deployed_or_running_deploy
 
   def create_deploy(user, attributes = {})
     before_command = attributes.delete(:before_command)
+    commit = attributes.delete(:commit)
     deploys.create(attributes.merge(release: !no_code_deployed, project: project)) do |deploy|
       commands = before_command.to_s.dup << script
       deploy.build_job(
-        project: project, user: user, command: commands, commit: deploy.reference, bypass_global_lock_check: true
+        project: project,
+        user: user,
+        command: commands,
+        commit: commit || deploy.reference,
```

Comment:
think you should assert that the reference resolves to the commit, otherwise we could get some weird situations where the commit is completely different...

Reply:
commits can be non-shas 😞 
old bug/feature